master/internal/command/notebook_manager.go

```diff
@@ -1,77 +1,97 @@
-// Package command provides utilities for commands.
 package command
 
 import (
-	"net/http"
-
-	"github.com/labstack/echo/v4"
-
-	"github.com/determined-ai/determined/master/internal/api"
-	"github.com/determined-ai/determined/master/internal/db"
-	"github.com/determined-ai/determined/master/internal/rm"
-	"github.com/determined-ai/determined/master/pkg/actor"
+	"github.com/determined-ai/determined/master/internal/task"
 	"github.com/determined-ai/determined/master/pkg/model"
+	"github.com/determined-ai/determined/master/pkg/protoutils"
 	"github.com/determined-ai/determined/proto/pkg/apiv1"
 	"github.com/determined-ai/determined/proto/pkg/notebookv1"
 )
 
-type notebookManager struct {
-	db *db.PgDB
-	rm rm.ResourceManager
+func (cs *commandService) GetNotebooks(req *apiv1.GetNotebooksRequest) (*apiv1.GetNotebooksResponse, error) {
+	cs.mu.Lock()
+	defer cs.mu.Unlock()
+
+	resp := &apiv1.GetNotebooksResponse{}
+	users := make(map[string]bool, len(req.Users))
+	for _, user := range req.Users {
+		users[user] = true
+	}
+	userIds := make(map[int32]bool, len(req.UserIds))
+	for _, user := range req.UserIds {
+		userIds[user] = true
+	}
+	for _, c := range cs.getAllNTSCType(model.TaskTypeNotebook) {
+		n := c.toNotebook()
+		if !((len(users) == 0 && len(userIds) == 0) || users[n.Username] || userIds[n.UserId]) {
+			continue
+		}
```

Comment:
can we possibly push this users/userIds agg + filtering by it into `getAllNTSCType` to reduce duplication?